### ISSUE TITLE
Experimental CLI features not working in WSL2

### ISSUE THREAD

**Issue by xJonathanLEI** [docker/for-win#9422]
- [x] I have tried with the latest version of my channel (**Stable** or ~Edge~)
  - [x] I have uploaded Diagnostics
  - Diagnostics ID: _84111AF6-E265-48F8-AB77-1B40092372F2/20201106153119_

### Steps to reproduce the behavior

I'm running Docker Desktop for Windows with the new WSL2 backend. I'm able to communicate with the same Docker daemon from both Windows (PowerShell) and WSL2 (Bash). This is really nice as I get to avoid the `chmod` problem when building images on Windows: I can just run `docker build` on WSL instead.

But then when I went ahead to play with experimental features I found an issue: the Docker CLI on the WSL side seems to ignore the `experimental` flag.

Experimental features are properly enabled in Docker Desktop settings:

![image](https://user-images.githubusercontent.com/19556359/98373658-8eded800-207a-11eb-8022-4519f58e3dff.png)

I can run experimental commands like `docker buildx` without issue on PowerShell:

    # Running from PowerShell
    $ docker buildx version
    github.com/docker/buildx v0.4.2-tp-docker fb7b670b764764dc4716df3eba07ffdae4cc47b2

The `experimental` flag is properly set on WSL:

    # Running from WSL
    $ sudo cat ~/.docker/config.json
    {
      "credsStore": "desktop.exe",
      "experimental": "enabled"
    }

Now run the same command on WSL:

    # Running from WSL
    $ sudo docker buildx version

### Expected behavior

The command should print the same result as running on Windows.

### Actual behavior

Output:

    docker: 'buildx' is not a docker command.
    See 'docker --help'

Tried setting `DOCKER_CLI_EXPERIMENTAL=enabled`, restarting WSL, restarting computer, no luck. The WSL-side CLI seems to completely ignore experimental features.

### Information
  - Windows Version: **10.0.19041 N/A Build 19041**
  - Docker Desktop Version: **2.5.0.0 (49427)**
  - Are you running inside a virtualized Windows e.g. on a cloud server or on a mac VM: **No**

**Comment by xJonathanLEI** [comment]
I figured it out.. For someone reading it from the future:

Somehow in WSL when you do:

    $ sudo docker xxxx

It reads from `/root/.docker/config.json` for configuration, instead of the home directory of the current user. I tried on a native Ubuntu host and it correctly reads from the current user even under `sudo`. It seems to be a bug but anyways...

So there're 2 possible solutions:

- Add your WSL user to `docker` group and execute Docker commands without `sudo`.
- Add the experimental flag to `/root/.docker/config.json` instead.

Both have been verified to work on my machine.
